engine/src/main/java/org/hibernate/search/v6poc/search/dsl/predicate/impl/MatchAllPredicateContextImpl.java

```diff
@@ -85,32 +83,25 @@ private MatchAllExceptContext getExceptContext() {
 		return exceptContext;
 	}
 
-	private class MatchAllExceptContext implements SearchPredicateDslContext<MatchAllPredicateContext<N>, CTX, C>,
-			SearchPredicateContributor<CTX, C> {
-
-		private final List<SearchPredicateContributor<CTX, ? super C>> children = new ArrayList<>();
+	private class MatchAllExceptContext implements SearchPredicateDslContext<MatchAllPredicateContext<N>, CTX, C> {
 
 		private final SearchPredicateContainerContextImpl<MatchAllPredicateContext<N>, CTX, C> containerContext;
 
+		private SearchPredicateContributor<CTX, ? super C> child;
```

Comment:
Callers can totally write `.matchAll().except(something).except(somethingElse).except(yetAnotherThing)`.

This should be a list. Or the enclosing class should hold a `List<MatchAllExceptContext>` instead of a single `MatchAllExceptContext`; whatever your prefer. The former would allow to add multiple clauses in the lambda syntax, which is nice.
In any case, only keeping the last `except` will not work.

I pushed an additional commit to your branch with a test.

Reply:
Done.